nwbwidgets/placefield.py

```diff
@@ -0,0 +1,321 @@
+
+
+## To-do
+# [X] Create PlaceFieldWidget class
+    # [X] Refactor place field calculation code to deal with nwb data type
+        # [X] Incorporate place field fxns into class
+        # [X] Change all internal attributes references
+        # [X]Change all internal method references
+
+    # [X] Get pos
+    # [X] Get time
+    # [X] Get spikes
+    # [] Get trials / epochs
+
+# [X] Submit draft PR
+
+    # [] 1D Place Field Widget
+        # [X] Incorporate nelpy package into widget
+        # [X] Speed threshold implementation
+        # [X] Add foreign group and sort controller to pick unit groups and ranges?
+        # [X] Normalized firing rate figure?
+        # [X] Add collapsed unit vizualization?
+        # [] Scale bar?
+        # [] Sort place cell tuning curves by peak firing rate position?
+        # [] Color palette control?
+
+# [] Dropdown that controls which unit
+
+# [x] Work in buttons / dropdowns / sliders to modify following parameters in place field calculation:
+# [] Different epochs
+# [x] Gaussian SD
+# [x] Speed threshold
+# [] Minimum firing rate
+# [] Place field thresh (% of local max)
+
+def route_placefield(spatial_series: pynwb.behavior.SpatialSeries):
+    print(spatial_series.data.shape)
+    if spatial_series.data.shape[1] == 2:
+        return PlaceFieldWidget(spatial_series)
+    elif spatial_series.data.shape[1] == 1:
+        return PlaceField_1D_Widget(spatial_series)
+    else:
+        print('Spatial series exceeds dimensionality for visualization')
+        return
+
+
+# Put widget rendering here
+class PlaceFieldWidget(widgets.HBox):
+
+    def __init__(self, spatial_series: pynwb.behavior.SpatialSeries, **kwargs):
+        super().__init__()
+
+
+        if hasattr(spatial_series.get_ancestor('NWBFile'),'velocity'):
+            velocity = spatial_series.get_ancestor('NWBFile').velocity
+        else:
+            velocity = []
+
+        self.units = spatial_series.get_ancestor('NWBFile').units
+        self.pos_tt = get_timeseries_tt(spatial_series)
+
+        istart = 0
+        istop = None
+
+        self.pos, self.unit = get_timeseries_in_units(spatial_series, istart, istop)
+
+        self.pixel_width = (np.nanmax(self.pos) - np.nanmin(self.pos)) / 1000
+
+        # Put widget controls here:
+        # - Minimum firing rate
+        # - Place field thresh (% of local max)
+
+        bft_gaussian = BoundedFloatText(value=0.0184, min=0, max=99999, description='gaussian sd (cm)')
+        bft_speed = BoundedFloatText(value=0.03, min=0, max=99999, description='speed threshold (cm/s)')
+        dd_unit_select = Dropdown(options=np.arange(len(self.units)), description='unit')
+        cb_velocity = Checkbox(value=False, description='use velocity', indent=False)
+
+        self.controls = dict(
+            gaussian_sd=bft_gaussian,
+            speed_thresh=bft_speed,
+            index=dd_unit_select,
+            use_velocity=cb_velocity,
+            velocity=fixed(velocity)
```

Comment:
```suggestion
```

Reply:
`fixed` can take a long time for some reason. Try to use class variables or `functools.partial` instead